zebra-test/src/transcript.rs

```diff
@@ -15,19 +15,25 @@ use tower::{Service, ServiceExt};
 
 type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
 
+/// An error-checking function: is the value an expected error?
```

Comment:
```suggestion
/// An error-checking function: is the value an expected error?
///
/// If the checked error is the expected error, the function should return `Ok(())`.
/// Otherwise, it should just return the checked error wrapped inside `Err`.
```
I was a little unsure of how it worked, so I wrote some extra documentation according to what I think is the expected implementation. I thought it might be a good way to verify my understanding and add some extra documentation :sweat_smile:

Reply:
Added as part of 568148d2

I also changed `Error` to `BoxError`, because that's what we use elsewhere in Zebra.